lib/cluster.js

```diff
@@ -496,6 +496,38 @@ Cluster.prototype.getInfoFromNode = function (redis, callback) {
   }, 1000));
 };
 
+Cluster.prototype.mastersExec = function () {
+  var masterNodes = _.values(this.masterNodes);
+
+  var argumentsLength = arguments.length;
+  var hasCallback = typeof arguments[argumentsLength - 1] === 'function';
+
+  var argsLength = hasCallback ? argumentsLength - 1 : argumentsLength;
+  var args = new Array(argsLength - 1);
+  for (var i = 1; i < argsLength; i++) {
+    args[i - 1] = arguments[i];
+  }
+
+  var command = arguments[0];
+  var promise = Promise.map(masterNodes, function sendCommandToRedis(redis) {
+    return redis[command].apply(redis, args);
+  });
+
+  if (hasCallback) {
+    return promise.nodeify(arguments[argsLength]);
+  }
+
+  return promise;
+}
+
+Cluster.prototype.flushCluster = function (callback) {
+  return this.mastersExec('flushall', callback);
+};
+
+Cluster.prototype.flushClusterDb = function (callback) {
+  return this.mastersExec('flushdb', callback);
```

Comment:
First idea seems cleaner, second would require a bit more code all the time. However we would want to make a single instance of a command distributor like this and I cant quite come up with the design just yet - do you have any particular implementation of this in mind?